AstarteDeviceSDKCSharp/Transport/MQTT/AstarteMqttTransport.cs

```diff
@@ -67,9 +67,9 @@ private async Task<IMqttClient> InitClientAsync()
             return mqttFactory.CreateMqttClient();
         }
 
-        private async Task CompleteAstarteConnection()
+        private async Task CompleteAstarteConnection(bool IsSessionPresent)
         {
-            if (!_introspectionSent)
+            if (!IsSessionPresent || !_introspectionSent)
             {
                 await SetupSubscriptions();
                 await SendIntrospection();
```

Comment:
For a complete `session present` handling, you need to resend all known properties from the device, so  you need to add the `ResendAllProperties` method here.

See see the java-sdk for reference: 
https://github.com/astarte-platform/astarte-device-sdk-java/blob/5b2f878dad8d455766ef209b81b5e1f101a438d8/DeviceSDK/src/main/java/org/astarteplatform/devicesdk/transport/mqtt/AstarteMqttV1Transport.java#L474

Reply:
Added method for sending properties from local storage to Astarte when session present is true.